test/unit-tests/debugger/attachDebugger.test.ts

```diff
@@ -0,0 +1,97 @@
+import * as assert from "assert";
+import * as vscode from "vscode";
+import * as lldb from "../../../src/debugger/lldb";
+import { attachDebugger } from "../../../src/commands/attachDebugger";
+import { mockNamespace } from "../MockUtils";
+import { mock, capture, instance, when, spy, verify, anything, deepEqual } from "ts-mockito";
+import { WorkspaceContext } from "../../../src/WorkspaceContext";
+
+suite("attachDebugger Unit Test Suite with Thenable Capture", () => {
+    const lldbSpy = spy(lldb);
+    const mockContext = mock(WorkspaceContext);
+    const windowMock = mockNamespace(vscode, "window");
+
+    setup(() => {});
+
+    test.skip("should call startDebugging with correct debugConfig and capture the return Thenable", async () => {
+        // Mock the list of processes returned by getLldbProcess
+        const processPickItems = [
+            { pid: 1234, label: "1234: Process1" },
+            { pid: 2345, label: "2345: Process2" },
+        ];
+        when(lldbSpy.getLldbProcess(anything())).thenResolve(processPickItems);
+
+        // Mock showQuickPick to return a selected process
+        when(
+            windowMock.showQuickPick(deepEqual(processPickItems)) as Promise<
+                (typeof processPickItems)[0]
+            >
+        ).thenResolve(processPickItems[0]);
+
+        // Spy on vscode.debug.startDebugging
+        const debugSpy = spy(vscode.debug);
+
+        // Call attachDebugger
+        await attachDebugger(instance(mockContext));
+
+        // Verify startDebugging was called once
+        verify(debugSpy.startDebugging(undefined, anything())).once();
+
+        // Now, capture the returned Thenable and assert its behavior
+        const capturedThenable = capture(debugSpy.startDebugging).last()[0];
+        assert.strictEqual(capturedThenable, true, "Thenable should resolve to true");
+    });
+});
+
+import * as sinon from "sinon";
```

Comment:
Can we accomplish what we need to with just ts-mockito? I'm hesitant to add multiple mocking libraries, especially when mocking should used as a be a last resort.

Reply:
Same wise, I was meant to have this patch up as a draft for discussion. For this test case, ts-mockito was not able to properly resolve the mock object.

Reply:
From what I have seen there should be a way to Proxy anything. It would be more of a discussion of we should use Sinon, getting rid of ts-mockito if necessary. This may fall into a case where we either test critical functionality at the integration level if there is no way to mock, or don't test at all if there is no way. We don't need to hit 100% coverage if there is too many hoops to jump through to get the source code under test

Reply:
Figured out a way to do it, just have to work around the overloaded signature and make sure things match.